### ISSUE TITLE
Deploy only goal for BW6 applications

### ISSUE THREAD

**Issue by suneelyaram** [TIBCOSoftware/bw6-plugin-maven#38]
Hi,

I would like to know which goal to use for Deployment only.

Goal should take EAR or Deployment Config as input and deploy it on target environment.

Is it available?

I am using 'deploy' goal but I have the following exception

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:2.7:deploy (default-deploy) on project tibco.bw.sample.binding.soap.http.BookStore.application.parent: Deployment failed: repository element was not specified in the POM inside distributionManagement element or in -DaltDeploymentRepository=id::layout::url parameter -> [Help 1]

**Comment by fanninj3** [comment]
Hi,

This is not currently available. Modifications are needed to the plugin, but it is possible.

Remember that Maven deploy goal is to deploy the created artifacts to a repository. You get that exception because you have not configured distributionManagement .

The closest you can get to a "BW6 deploy" is to use the INSTALL goal, but this will build a fresh EAR an then deploy to a BW6 domain.

**Comment by suneelyaram** [comment]
@fanninj3 thank you for your information. May I please know when could we expect this feature?

Could you kindly also check if the following is possible at the moment?
https://github.com/TIBCOSoftware/bw6-plugin-maven/issues/36

ATM, I am using something as below. But it doesn't seems to work for EAR files
`<build>  
  <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-jar-plugin</artifactId>
        <version>2.1</version>
        <configuration>
          <archive>
            <manifest>
              <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
            </manifest>
            <manifestEntries>
              <Implementation-Build>${buildNumber}</Implementation-Build>              
            </manifestEntries>
          </archive>
        </configuration>
      </plugin>
    </plugins>
 </build>   
`

**Comment by fanninj3** [comment]
Hi,
I cannot check into this, but it would be very nice to have this feature included. Especially with easy support for external profile file.

**Comment by suneelyaram** [comment]
@RajibBiswasTibco may I please know, if this enhancement is already planned for release? 
Thanks in advance!

**Comment by edlegaspi** [comment]
Hi @RajibBiswasTibco, I'm also curious as to when this enhancement would be included in the plug-in. Would you be able to give an estimate for its release? Thanks in advance!

**Comment by RajibBiswasTibco** [comment]
Hi @suneelyaram , @edlegaspi , this feature is not yet scheduled. Would you like to contribute on this feature?

**Comment by Tantino** [comment]
hi @RajibBiswasTibco 
I like to contribute on this feature.
thanks and regards.

**Comment by akpuvvada** [comment]
I would like to know when this feature is available.

**Comment by phenelle** [comment]
Hi guys, 
I was able to only deploy using the following command within the bwapp folder:

`mvn bw6maven:bwinstall`

**Comment by Lolek1978** [comment]
hi
How did you do it? 
What version of plugin did you use?

**Comment by phenelle** [comment]
I think it was 1.2.1 or 1.2.2.
It requires little config I guess. (if you are able to do mvn install, then for sure you'll be able to mvn bw6maven:bwinstall)
Those goals are defined somewhere within the pluging code..

**Comment by Lolek1978** [comment]
Thx for info :) - i managed with this:). 

The second issue is how to pass to "mvn bw6maven:bwinstall" location for our ears file.

Default location is "target" folder in our project folder.
In debug mode during maven call I saw te variables like 
<outputDirectory>${project.build.directory}</outputDirectory> and i wonder how to pass this variable.

I tried with -Dproject.build.directory=dirWithMyEars but it didn't work it.

I need to pass specific location for my ears to mvn for execute.


Any ideas?
